apps/explorer/src/components/GasPriceCard/index.tsx

```diff
@@ -0,0 +1,246 @@
+import { SUI_TYPE_ARG } from '@mysten/sui.js';
+import { useQuery } from '@tanstack/react-query';
+import { ParentSize } from '@visx/responsive';
+import clsx from 'clsx';
+import { useMemo, useState } from 'react';
+
+import { ErrorBoundary } from '../error-boundary/ErrorBoundary';
+import { Graph, isDefined } from './Graph';
+import { type EpochGasInfo } from './types';
+
+import { Card } from '~/ui/Card';
+import { FilterList } from '~/ui/FilterList';
+import { Heading } from '~/ui/Heading';
+import { ListboxSelect } from '~/ui/ListboxSelect';
+import { LoadingSpinner } from '~/ui/LoadingSpinner';
+import { Stats } from '~/ui/Stats';
+import { Text } from '~/ui/Text';
+
+const UNITS = ['MIST', 'SUI'] as const;
+type UnitsType = (typeof UNITS)[number];
+const GRAPH_DURATIONS = ['7 EPOCHS', '30 EPOCHS'] as const;
+type GraphDurationsType = (typeof GRAPH_DURATIONS)[number];
+const GRAPH_DURATIONS_MAP: Record<GraphDurationsType, number> = {
+    '7 EPOCHS': 7,
+    '30 EPOCHS': 30,
+};
+
+function useHistoricalGasPrices() {
+    const rpc = useRpcClient();
+    return useQuery<EpochGasInfo[]>(
+        ['get', 'last 30 epochs gas price'],
+        async () => {
+            // TODO: update this to get the gas price from the epoch itself rather than the previous one
+            // once this is deployed https://github.com/MystenLabs/sui/pull/11388
+            // every epoch contains the gas price for the next one
+            const epochs = (
+                await rpc.getEpochs({
+                    descendingOrder: true,
+                    limit: 31,
+                })
+            ).data.reverse();
+            // remove the current epoch since it would have the gasPrice for the next one
+            epochs.pop();
+            return epochs.map((anEpoch) => ({
+                epoch: Number(anEpoch.epoch) + 1,
+                referenceGasPrice: anEpoch.endOfEpochInfo?.referenceGasPrice
+                    ? BigInt(anEpoch.endOfEpochInfo?.referenceGasPrice)
+                    : null,
+                date: anEpoch.endOfEpochInfo?.epochEndTimestamp
+                    ? new Date(
+                          Number(anEpoch.endOfEpochInfo?.epochEndTimestamp)
+                      )
+                    : null,
+            }));
+        }
+    );
+}
+
+function useGasPriceAverage(totalEpochs: number) {
+    const { data, isLoading } = useHistoricalGasPrices();
+    const average = useMemo(() => {
+        const epochs = data?.slice(-totalEpochs) || [];
+        const epochsWithPrices = epochs.filter(
+            ({ referenceGasPrice }) => referenceGasPrice !== null
+        );
+        if (epochsWithPrices.length) {
+            const sum = epochsWithPrices.reduce(
+                (acc, { referenceGasPrice }) =>
+                    acc + BigInt(referenceGasPrice!),
+                0n
+            );
+            return sum / BigInt(epochsWithPrices.length);
+        }
+        return null;
+    }, [data, totalEpochs]);
```

Comment:
nit 
const data= useHistoricalGasPrices();
...
`return {
       ...data,
        data: average
     };`

Reply:
yeah good point, I just wanted to use only loading but I think you are right this is a better pattern!